### PR TITLE
Normalization: Upgrade MySQL to dbt 1.0.0

### LINKED ISSUE #9484: 🐛 Destination MySQL: database argument missing at connection time
## Environment
- **Airbyte version**:  0.35.4-alpha
- **OS Version / Instance**: Windows 10 Pro
- **Deployment**: Docker
- **Destination Connector and version**: Mysql 0.1.15
- **Severity**: Critical
- **Step where error happened**: Sync job

## Current Behavior
During normalization DBT can't connect to the targetted database because of `_mysql_connector.MySQLInterfaceError: No database selected` error.
Replication of raw data works as expected.

## Expected Behavior
The normalization process should run flawlessly.

## Logs

[sync logs](https://github.com/airbytehq/airbyte/files/7864610/logs-7-2.txt)
[dbt SQL model](https://github.com/airbytehq/airbyte/files/7864621/orb_job.txt)

## Steps to Reproduce
1. Set up a connection with MySQL destination and enable basic normalization
2. Run the sync

### PR DESCRIPTION
**Pull request by alafanechere**
## What
`dbt-mysql` now supports DBT 1.0.0.
Upgrading to 1.0.0 can fix some existing bug such as  https://github.com/airbytehq/airbyte/issues/9484

## How
* Edit `mysql.Dockerfile`
* Use `event-buffer-size` in test for `normalization-mysql`
* Bump normalization version

### CODE REVIEW

**Comment by ChristopheDuong** [comment]
You might need to tweak the `dbt_project.yml` files as it was done here for this upgrade: https://github.com/airbytehq/airbyte/pull/11051

**Review by ChristopheDuong — CHANGES_REQUESTED** [review]
To upgrade to dbt 1.0.0, you might need to tweak the dbt_project.yml and package.yml files as it was done in the PR https://github.com/airbytehq/airbyte/pull/11051

(maybe check if you need to regenerate output tests files too)

**Comment by ChristopheDuong** [comment]
It seems the failing tests may be related to https://github.com/airbytehq/airbyte/pull/11267#discussion_r838403965  maybe?

**Review by ChristopheDuong — APPROVED** [review]
Could you check the event-buffer-size option issue from https://github.com/airbytehq/airbyte/pull/11578 while you are publishing this please?

**Comment by alafanechere** [comment]
@ChristopheDuong the test ^ failed because the build of `destination-mssql` image failed:
```
#8 [build 4/4] RUN tar xf destination-mssql.tar --strip-components=1 && rm -rf destination-mssql.tar
#8 sha256:6aef1e7c3a165e325a1e5265e05c5ece72ce34711b89d8743a4a14875268dc37
#8 0.408 tar: destination-mssql.tar: Cannot open: No such file or directory
#8 0.408 tar: Error is not recoverable: exiting now
#8 ERROR: executor failed running [/bin/sh -c tar xf $***APPLICATION***.tar --strip-components=1 && rm -rf $***APPLICATION***.tar]: exit code: 2
```

Do you have an idea for the root cause?

**Comment by alafanechere** [comment]
@ChristopheDuong now tests are failing for a more intelligible reason:
```
 AssertionError: docker run --rm --init -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_simple_streams:/workspace -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_simple_streams/build:/build -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_simple_streams/logs:/logs -v /tmp:/tmp --network host --entrypoint /usr/local/bin/dbt -i airbyte/normalization-mysql:dev --event-buffer-size=10000 run --profiles-dir=/workspace --project-dir=/workspace --full-refresh
	 E         	terminated with return code 1 with 5 'Error/Warning/Fail' mention(s).
	 E       assert 5 == 0
	 E         +5
	 E         -0
```

**Comment by ChristopheDuong** [comment]
the useful eror in the log would be
```
 	 08:24:58  Finished running 6 view models, 14 table models in 1.23s.
	 08:24:58  
	 08:24:58  Completed with 5 errors and 0 warnings:
	 08:24:58  
	 08:24:58  Database Error in model 1_prefix_startwith_number_scd (models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/1_prefix_startwith_number_scd.sql)
	 08:24:58    1292 (22007): Truncated incorrect DOUBLE value: '-'
	 08:24:58    compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/1_prefix_startwith_number_scd.sql
	 08:24:58  
	 08:24:58  Database Error in model dedup_cdc_excluded_scd (models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/dedup_cdc_excluded_scd.sql)
	 08:24:58    1292 (22007): Truncated incorrect DOUBLE value: '-'
	 08:24:58    compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/dedup_cdc_excluded_scd.sql
	 08:24:58  
	 08:24:58  Database Error in model dedup_exchange_rate_scd (models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/dedup_exchange_rate_scd.sql)
	 08:24:58    1292 (22007): Truncated incorrect DOUBLE value: '-'
	 08:24:58    compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/dedup_exchange_rate_scd.sql
	 08:24:58  
	 08:24:58  Database Error in model multiple_column_names_conflicts_scd (models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/multiple_column_names_conflicts_scd.sql)
	 08:24:58    1292 (22007): Truncated incorrect DOUBLE value: '-'
	 08:24:58    compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/multiple_column_names_conflicts_scd.sql
	 08:24:58  
	 08:24:58  Database Error in model pos_dedup_cdcx_scd (models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/pos_dedup_cdcx_scd.sql)
	 08:24:58    1292 (22007): Truncated incorrect DOUBLE value: '-'
	 08:24:58    compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/scd/test_normalization_ci_txjrr/pos_dedup_cdcx_scd.sql
	 08:24:58  
	 08:24:58  Done. PASS=9 WARN=0 ERROR=5 SKIP=6 TOTAL=20
```

**Comment by edgao** [comment]
for posterity: https://github.com/dbt-labs/dbt-utils/pull/373 broke mysql output; latest commit on this branch adds an override for mysql's concat logic.

test output has some odd capitalization diffs, which is weird but probably fine? (since mysql tablenames are case-insensitive)

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_nested_streams/first_output/airbyte_incremental/test_normalization/nested_stream_with_co___names_partition_data.sql [hunk 1]
```diff
@@ -20,7 +20,7 @@ with numbers as (
     select
 
     
-    p0.generated_number * pow(2, 0)
+    p0.generated_number * power(2, 0)
```
Comment (edgao) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_nested_streams/first_output/airbyte_incremental/test_normalization/nested_stream_with_co___names_partition_data.sql:23]: `pow` and `power` are apparently equivalent https://stackoverflow.com/a/51615601

**Comment by edgao** [comment]
tests failed due to https://github.com/airbytehq/airbyte/pull/9610 again (not sure why the comment didn't get edited though). Waiting for the test run on _that_ pr to finish, and will rerun tests here afterward. Realistically, I probably won't do that until tomorrow though.

**Comment by grubberr** [comment]
> @ChristopheDuong the test ^ failed because the build of `destination-mssql` image failed:
> 
> ```
> #8 [build 4/4] RUN tar xf destination-mssql.tar --strip-components=1 && rm -rf destination-mssql.tar
> #8 sha256:6aef1e7c3a165e325a1e5265e05c5ece72ce34711b89d8743a4a14875268dc37
> #8 0.408 tar: destination-mssql.tar: Cannot open: No such file or directory
> #8 0.408 tar: Error is not recoverable: exiting now
> #8 ERROR: executor failed running [/bin/sh -c tar xf $***APPLICATION***.tar --strip-components=1 && rm -rf $***APPLICATION***.tar]: exit code: 2
> ```
> 
> Do you have an idea for the root cause?

I saw this problem a lot of time in past 
I suspect that it's some concurrency side effect when multiple gradle tasks running in parrallel 
( but to be honest I don't have 100% proves for this)

yesterday to work-around this problem in this PR https://github.com/airbytehq/airbyte/pull/9610
I have done this temp commit 
https://github.com/airbytehq/airbyte/pull/9610/commits/1c48d6e3e1de8b59958a275d8006a86dc7a1e37d

and it helped

**Comment by alafanechere** [comment]
Thanks @grubberr, let's give this a try 😄

**Comment by alafanechere** [comment]
@edgao does it mean we need wait for https://github.com/airbytehq/airbyte/pull/9610 to be merged?

**Comment by edgao** [comment]
sort of. Every time that PR gets `/test`-ed, it'll break tests everywhere else. I cleaned up our cluster again, so tests should be fixed here, hopefully

**Comment by grubberr** [comment]
@alafanechere I see that our `base-normalization` tests are not good isolated, please do before running tests:

```
psql -h redshift-cluster-2.c5lzdndklo9c.us-east-2.redshift.amazonaws.com -p 5439 -d integrationtests --user awsuser

grubberr_test=# drop database integrationtests;
DROP DATABASE
grubberr_test=# create database integrationtests;
CREATE DATABASE
```

**Comment by edgao** [comment]
hm, the last run still failed with the `tar: destination-mssql.tar: Cannot open: No such file or directory` error. gonna just retry it

**Comment by edgao** [comment]
rebumped version to 0.1.77 and reattempting the publish

**Comment by alafanechere** [comment]
@edgao is now failing with Snowflake 🤦 : 
`Operation failed because soft limit on objects of type 'Column' per table was exceeded. Please reduce number of 'Column's or contact Snowflake support about raising the limit.`
Do you think this is transient?

**Comment by edgao** [comment]
that's just the logs being misleading. The real error is even higher:
```
	 Unable to find image 'airbyte/destination-bigquery:dev' locally
	 docker: Error response from daemon: manifest for airbyte/destination-bigquery:dev not found: manifest unknown: manifest unknown.
```
this is definitely transient; retrying.

I should put this in the readme later, but the way to find the "actual" error message:
1. Find the test summary (it's a red line of text, e.g. `============= 1 failed, 23 passed, 9 skipped in 2233.36s (0:37:13) =============`)
2. Above it, there are some lines that start with `FAILED`, e.g. `AILED integration_tests/test_ephemeral.py::test_destination_supported_limits[DestinationType.BIGQUERY-1000]`
3. Search the logs for the first occurrence of the text inside the brackets (e.g. `DestinationType.BIGQUERY-1000`)
4. Scroll down until you find the error. (in this specific example, it's kind of weird because it somehow got interleaved with a bunch of gradle messages so you have to scroll quite a long distance, but normally it's pretty close)

**Comment by alafanechere** [comment]
@edgao I re-ran the test and found the following errors:
**MSSQL**
```ERROR i.a.i.b.FailureTrackingAirbyteMessageConsumer(start):39 - Exception while starting consumer
	 java.sql.SQLException: Cannot create PoolableConnectionFactory (Cannot open database "test_ephemeral" requested by the login. The login failed. ClientConnectionId:f0df6262-7a8b-4d03-9688-c1eacbf22507)
```

Do you think this is transient? Is the `test_ephemeral` DB created on test run?


**REDSHIFT** 
```
Error during updateVarcharDataColumnToSuperDataColumn() appears: 
	 java.sql.SQLException: [Amazon](500310) Invalid operation: cannot drop table _airbyte_raw_stream_with_1000_columnsckjyv column _airbyte_data because other objects depend on it 
	 Hint: Use DROP ... CASCADE to drop the dependent objects too.
```

Do you think this is because we need to clean some other test-related table from other test runs?


PS: thank you for explaining the process of spotting the right error message 🙏

**Comment by grubberr** [comment]
@alafanechere about REDSHIFT I see that base-normalization test are not isolated
to make them PASS tyou need to re-create `integrationtests ` database 

```
psql -h redshift-cluster-2.c5lzdndklo9c.us-east-2.redshift.amazonaws.com -p 5439 -d integrationtests --user awsuser

drop database integrationtests;
create database integrationtests;
```

**Comment by edgao** [comment]
how does this PR keep running into different errors :joy:

that _could_ be transient? The MSSQL test creates its own database, so it should just work.... retrying the test run.

Also, merged in master to get the changes from https://github.com/airbytehq/airbyte/pull/12064, which should fix the redshift normalization test.

**Comment by grubberr** [comment]
@edgao when you run 
`/test connector=bases/base-normalization`
you run normalization tests over all possible destinations we have

**Comment by alafanechere** [comment]
> @edgao when you run `/test connector=bases/base-normalization` you run normalization tests over all possible destinations we have

Do we have a way to only run tests for Mysql normalization?

**Comment by edgao** [comment]
@alafanechere fyi re: current test failure: discussion here https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1650550398654509?thread_ts=1650534020.615949&cid=C02U9R3AF37 (apparently our formatter is forcing uppercase `REF`, but dbt requires lowercase `ref`)

**Comment by alafanechere** [comment]
`docker: Error response from daemon: manifest for airbyte/destination-bigquery:dev not found: manifest unknown: manifest unknown.`

**Comment by edgao** [comment]
I think we are still running into the REF error, also.

But that's pretty weird; I suspect there's a race condition in the build somewhere, but haven't been able to track it down. That error has been transient the other couple times I've seen it.

**Comment by grubberr** [comment]
I see that problem with `REF` still not fixed
https://github.com/airbytehq/airbyte/runs/6136132925?check_suite_focus=true#step:17:20010

**Comment by edgao** [comment]
nice, that means the CI script hack is working (or is at least not breaking anything) :P

but yeah, I think the `REF` thing is going to block publishing base-normalization until it's fixed

**Comment by alafanechere** [comment]
@grubberr / @edgao do we have an issue for the `REF` thing?

**Comment by droxer** [comment]
I encountered the basic normalization issue for mysql 8 destination.  And the issue can be fixed by upgrading dbt mysql 1.0.0 . do you mind I create another PR for the issue, or you will be keep working this https://github.com/airbytehq/airbyte/pull/11470 in next few days?

**Comment by grubberr** [comment]
> @grubberr / @edgao do we have an issue for the `REF` thing?

I have created one https://github.com/airbytehq/airbyte/issues/12384

**Comment by alafanechere** [comment]
Hi @droxer, we are currently trying to upgrade the MySQL adapter to 1.0.0. We are currently blocked by #12384 but it's on it's way.

**Comment by tuliren** [comment]
The MySQL destination acceptance test has been broken for at least the last five days due to normalization. It's unclear how long it had been failing. So I triggered one to see if the new dbt version fixes this problem.

**Comment by DoNotPanicUA** [comment]
Hi @tuliren ,
Just want to let you know that MySql DAT is now fixed. There was an issue with the normalization but now it's fixed by #12924

**Comment by edgao** [comment]
going to publish skipping tests. I ran the mysql normalization tests locally and they passed.

**Pull request merged**